plugins/common/tls/config_test.go

```diff
@@ -42,6 +42,34 @@ func TestClientConfig(t *testing.T) {
 				TLSKeyPwd: "",
 			},
 		},
+		{
+			name: "fail with enc tls key",
+			client: tls.ClientConfig{
+				TLSCA:     pki.CACertPath(),
+				TLSCert:   pki.ClientCertPath(),
+				TLSKey:    pki.ClientEncKeyPath(),
+				TLSKeyPwd: "changeme",
+			},
+			expNil: true,
+			expErr: true,
+		},
+		{
+			name: "success with cert and key",
+			client: tls.ClientConfig{
+				TLSCA:         pki.CACertPath(),
+				TLSCertAndKey: pki.ClientCertAndKeyPath(),
+			},
+		},
+		{
+			name: "fail with cert and enc key",
+			client: tls.ClientConfig{
+				TLSCA:         pki.CACertPath(),
+				TLSCertAndKey: pki.ClientCertAndEncKeyPath(),
+				TLSKeyPwd:     "changeme",
+			},
+			expNil: true,
+			expErr: true,
```

Comment:
Similar to line 53-54

Reply:
Shouldn't this also not be nil and not have an error?

Reply:
@bustedware see this question about the error value.